### PR TITLE
[docs] Relative filenames. Using doxygen-awesome 

### PR DESCRIPTION
**Pull request by 1uc**
Improves the Doxygen generated documentation with the following features.

- [x] Adds the doxygen-awesome style sheet.
- [x] Makes filenames relative, e.g. in file dependency graphs.

Improves #472

### CODE REVIEW

File: doc/doxygen-awesome-css/download_stylesheet.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#! /usr/bin/env bash
```
Comment (tdegeus) [inline doc/doxygen-awesome-css/download_stylesheet.sh:1]: Why is this file here? Wouldn't it make sense to include all needed files here (that is at least what I do elsewhere)?

**Review by tdegeus — COMMENTED** [review]
Awesome! I was tempted to do this already when I looked at the output in your other issue.

File: doc/doxygen-awesome-css/download_stylesheet.sh (hunk 1 above)
Comment (1uc) [reply doc/doxygen-awesome-css/download_stylesheet.sh:1]: All files needed are included in the repository. This file can be removed.

The hope is that this file documents which files need to be downloaded to update the styling. The script provides the answer, just one, and it also provides the URL where that one file can be found (hopefully).

Would you like me to remove it?

Comment (tdegeus) [reply doc/doxygen-awesome-css/download_stylesheet.sh:1]: My suggestion would be to have a GH Action check if doxygen-awesome is outdated. It could run for example on all merges to the main branch. 
Ideally it could even open a PR if it founds things that are outdated, but I don't know if that is possible.

Comment (tdegeus) [reply doc/doxygen-awesome-css/download_stylesheet.sh:1]: Just the first hit when searching : https://github.com/istein/gh-actions-open-pr/blob/main/.github/workflows/pr.yml 
It seems like we can modify this to do what we want

Comment (1uc) [reply doc/doxygen-awesome-css/download_stylesheet.sh:1]: Nice idea, I'll improve the PR to incorporate this.

Comment (tdegeus) [reply doc/doxygen-awesome-css/download_stylesheet.sh:1]: Amazing. Let me know if you need help!

**Comment by 1uc** [comment]
I've tested the workflow in a separate repository quite thoroughly. Is there a way of testing it in this repo without effecting others adversely?

File: .github/workflows/check_doxygen_awesome_version.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+on:
+  workflow_dispatch:
+
+jobs:
+  check-for-updates:
+    runs-on: ubuntu-20.04
```
Comment (tdegeus) [inline .github/workflows/check_doxygen_awesome_version.yml:6]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by tdegeus — CHANGES_REQUESTED** [review]
Other than the minor comment to me it looks fine. 
No, I don't know if we can pre-test it. I think we will just have to try. I will try it elsewhere too, just to verify.

File: .github/workflows/check_doxygen_awesome_version.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+on:
```
Comment (tdegeus) [inline .github/workflows/check_doxygen_awesome_version.yml:1]:
```suggestion
name: Auto-update doxygen-awesome

on:
```

**Comment by tdegeus** [comment]
I check too, works like a charm! https://github.com/tdegeus/GooseFEM

**Review by tdegeus — APPROVED** [review]

**Comment by 1uc** [comment]
It worked :) I was not expecting it to trigger this soon. Very nice idea of yours. I'll look into turning it into an GH action. Thanks for all the feedback during review.

**Comment by tdegeus** [comment]
Like a charm! Thanks again, this is a really nice feature. It would be really nice to generalise this!
